server/accounts.go

```diff
@@ -660,9 +664,31 @@ func (a *Account) checkActivation(acc *Account, claim *jwt.Import, expTimer bool
 			})
 		}
 	}
+
 	return true
 }
 
+// Returns true if the activation claim is trusted. That is the issuer matches
+// the account or is an entry in the signing keys.
+func (a *Account) isIssuerClaimTrusted(claims *jwt.ActivationClaims) bool {
+	// if no issuer account, issuer is the account
+	if claims.IssuerAccount == "" {
+		return true
+	}
+	// get the referenced account
+	if a.srv != nil {
+		ia, err := a.srv.lookupAccount(claims.IssuerAccount)
+		if err != nil {
+			return false
+		}
+		// get the signing key - if it doesn't exist, reject
+		_, ok := ia.signingKeys[claims.Issuer]
```

Comment:
Think map may be overkill, if small list and we just walk and compare like operators will be faster.

Reply:
fixed